### ISSUE TITLE
Support for parrallel jobs

### ISSUE THREAD

**Issue by varuzam** [actions/runner#794]
**Describe the enhancement**
So far the runner runs only one job. It would be nice to have a configuration parameter for max parallel jobs

**Comment by ad-m** [comment]
The workaround for this problem is to install and run the agent multiple times in different subdirectories. In our case, this allows us to increase the use of resources as our tasks are not intensive (API requests to external services), similar to Terraform.
Repeatedly running an agent has a negative impact on memory usage and creates management problems when adding and removing a worker. We use 10 workers per 1 virtual machine (1vCPU, 1 GB RAM, 3 GB SWAP), so 10x increase in density.
In our case, we use the Docker cache instead of the GitHub Actions cache. However, it should be noted that multiple installations also limit the cache efficiency, and consequently may affect disk space usage (potentially, this can be circumvented by appropriate symlinks).

**Comment by timharris777** [comment]
We would like this as well. The option to run multiple self-hosted-runners is fine, but it would be nice to define how many parrallel/concurrent jobs could be run at once by a single runner.

**Comment by timharris777** [comment]
++ @jeff-miller-cfa, thought you might be interested

**Comment by marcinz** [comment]
Are there instructions anywhere on how to increase parallelism with self-hosted runners? Do I have to basically start multiple runners and add them separately? That seems a little tedious.

**Comment by timharris777** [comment]
> Are there instructions anywhere on how to increase parallelism with self-hosted runners? Do I have to basically start multiple runners and add them separately? That seems a little tedious.

Pretty much

**Comment by marcinz** [comment]
Thanks @timharris777. I wonder if Github does not care about self hosted runners, or what is the reason for this limitation. Coming from Gitlab CI, this just seems pretty bad.

**Comment by stealthHat** [comment]
any update about this issue?

**Comment by ethomson** [comment]
The runner was architected in such a way that we run a single job per runner process, and an individual runner has state within an individual directory on the filesystem.  Our guidance would be to run multiple runners in separate directories.

An architectural change here to support multiple jobs by a single runner would be a significant investment, and one that's rather different than the way we're looking at parallelism (which is Kubernetes).

If you want a few runners in parallel, we'd encourage you to run multiple jobs in separate directories.  If you want a _lot_ of runners in parallel, we'd encourage you to orchestrate them with Kubernetes.
